byterun/caml/misc.h

```diff
@@ -124,7 +124,94 @@ CAMLnoreturn_end;
 CAMLextern char * caml_strdup(const char * s);
 CAMLextern char * caml_strconcat(int n, ...); /* n args of const char * type */
 
-/* <private> */
+/* Use macros for some system calls being called from OCaml itself.
+  These calls can be either traced for security reasons, or changed to
+  virtualize the program. */
+
+#ifndef CAML_WITH_CPLUGINS
+
```

Comment:
Then you have two different features: plug-ins and system-call interception. You are pushing the latter is a security feature, so on Windows you have to either make sure it works or disable it entirely. Leaving it half-working is unacceptable, it's way worse than not having the feature at all.